### PR TITLE
Feature/ch16909/drop canner users

### PR DESCRIPTION
**Pull request by aar0nTw**
** PR checklist **

- [x] Ensure you have added or ran the appropriate tests for your PR.
- [x] DCO signed

**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
```

### CODE REVIEW

File: packages/client/src/admin/User/UserDetail.tsx [hunk 1]
```diff
@@ -1,44 +1,315 @@
...
 
 const { TabPane } = Tabs;
 
+const groupColumns = [
+  {
+    title: 'Name',
+    dataIndex: 'name',
+    render: (text, record, cannerProps) => {
+      const onClick = () => {
+        // /group/record.id
```
Comment (neighborhood999) [inline packages/client/src/admin/User/UserDetail.tsx:29]: Forgot to add logic?

File: packages/client/src/admin/User/UserDetail.tsx [hunk 2]
```diff
@@ -1,44 +1,315 @@
...
+import CheckableInputNumber from 'cms-components/customize-number-checkbox';
+import CustomRelationTable from '../share/RelationTable';
+import { User, UpdateUser, UserGroups } from 'queries/User.graphql';
+import { errorHandler } from 'utils/errorHandler';
 
 const { TabPane } = Tabs;
 
+const groupColumns = [
+  {
+    title: 'Name',
```
Comment (neighborhood999) [inline packages/client/src/admin/User/UserDetail.tsx:25]: column need to add `key`

File: packages/client/src/admin/share/RelationPicker.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        onChange={this.handleTableChange}
+        columns={columns.map((column: Record<string, any>) => {
+          if (column.sorter && column.dataIndex === sorter.field) {
+            column.sortOrder = sorter.order;
+          } else {
+            column.sortOrder = false;
+          }
+          return column;
+        })}
+        // $FlowFixMe
```
Comment (neighborhood999) [inline packages/client/src/admin/share/RelationPicker.tsx:153]: Remove it?

**Comment by aar0nTw** [comment]
@neighborhood999 removed useless canner schema already.

File: packages/client/src/admin/User/UserAdd.tsx [hunk 4]
```diff
@@ -1,42 +1,188 @@
...
+export const UserAdd = compose(
+  Form.create(),
+  withRouter,
+  graphql(CreateUser, {
+    name: 'createUser',
+    alias: 'withCreateUser',
+    options: (props: any) => ({
+      onCompleted: (data: any) => {
+        const { history } = props;
+        history.push({
```
Comment (neighborhood999) [inline packages/client/src/admin/User/UserAdd.tsx:167]: `history.push(/* .. */)`

File: packages/client/src/admin/User/UserList.tsx [hunk 5]
```diff
@@ -1,187 +1,271 @@
...
-          data-testid="edit-button"
+        <Button
+          icon={'edit'}
+          data-testid='edit-button'
           onClick={() => edit(record.id)}
-        >
-        </Button>
-        <Button icon="delete"
-          data-testid="delete-button"
+        ></Button>
```
Comment (neighborhood999) [inline packages/client/src/admin/User/UserList.tsx:81]:
```suggestion
        />
```

File: packages/client/src/admin/share/RelationTable.js [hunk 6]
```diff
@@ -0,0 +1,135 @@
+import React, { PureComponent } from 'react';
+import { Icon, Table, Button } from 'antd';
+import { remove } from 'lodash';
```
Comment (neighborhood999) [inline packages/client/src/admin/share/RelationTable.js:3]:
```suggestion
import { remove, difference } from 'lodash';
```

and remove below `difference` import

File: packages/client/src/admin/share/RelationTable.js [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+  showModal = () => {
+    this.setState({
+      modalVisible: true,
+    });
+  };
+
+  handleOk = (queue, originData) => {
+    let { onChange, refId, value = [] } = this.props;
+    // $FlowFixMe
```
Comment (neighborhood999) [inline packages/client/src/admin/share/RelationTable.js:30]: remove comment

File: packages/client/src/admin/share/RelationTable.js [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+          dataSource={showValues}
+          columns={columns}
+          style={{ marginBottom: 16 }}
+        />
+        {!disabled && modalVisible && (
+          <RelationPicker
+            visible={modalVisible}
+            onOk={this.handleOk}
+            onCancel={this.handleCancel}
+            // $FlowFixMe
```
Comment (neighborhood999) [inline packages/client/src/admin/share/RelationTable.js:122]: remove comment

File: packages/client/src/admin/User/UserDetail.tsx [hunk 9]
```diff
@@ -1,44 +1,382 @@
...
+    alias: 'withUpdateUser',
+    options: (props: any) => ({
+      variables: {
+        where: {
+          id: props.match.params.id,
+        },
+      },
+      onCompleted: (data: any) => {
+        const { history } = props;
+        history.push({
```
Comment (neighborhood999) [inline packages/client/src/admin/User/UserDetail.tsx]: here

File: packages/client/src/cms-toolbar/text.js [hunk 10]
```diff
@@ -22,7 +22,8 @@ export default class TextFilter extends Component {
       onChange({
         [name]: {
           contains: value
-        }
+        },
+        [name]: value
```
Comment (neighborhood999) [inline packages/client/src/cms-toolbar/text.js:26]: remove it

**Review by neighborhood999 — APPROVED** [review]

**Pull request merged**